packages/zen-mapper/src/zen_mapper/test_cover.py

```diff
@@ -126,3 +131,107 @@ def test_width_balanced_int():
     cover_scheme = Width_Balanced_Cover(3, 0.4)
     covered_data = reduce(lambda acc, new: acc.union(new), cover_scheme(data), set())
     assert len(data) == len(covered_data)
+
+
+def test_data_balanced_simple():
+    """
+    Test that Data_Balanced_Cover partitions data as expected for a tiny array.
+    """
+    data = np.array([10, 20, 30, 40, 50])
+    cover = Data_Balanced_Cover(2, 0.5)
+    groups = cover(data)
+
+    assert len(groups) == 2
+    all_points = reduce(lambda acc, g: acc.union(g), map(set, groups), set())
+    assert all_points == set(range(len(data)))
+    overlap = set(groups[0]).intersection(groups[1])
+    assert len(overlap) > 0
+
+
+def test_data_balanced_complete_coverage_random():
+    """Ensure that for arbitrary random data all indices are covered."""
+    rng = np.random.default_rng(42)
+    data = rng.normal(size=200)
+    n_elements = 5
+    overlap = 0.3
+
+    cover = Data_Balanced_Cover(n_elements, overlap)
+    groups = cover(data)
+
+    all_indices = set(np.concatenate(groups))
+    assert all_indices == set(range(len(data)))
+
+
+def test_data_balanced_small_dataset_error():
+    """Should raise ValueError if data has fewer points than groups."""
+    data = np.arange(3)
+    error_raised = False
+    try:
+        Data_Balanced_Cover(5, 0.4)(data)
+    except ValueError:
+        error_raised = True
+    assert error_raised
+
+
+def test_data_balanced_invalid_overlap():
+    """Should reject invalid percent_overlap."""
+    for overlap in [0, 1, -0.1, 2.0]:
+        error_raised = False
+        try:
+            Data_Balanced_Cover(3, overlap)
+        except ValueError:
+            error_raised = True
+        assert error_raised
```

Comment:
```py
@pytest.mark.parameterize("overlap", [0, 1, -0.1, 2.0])
def test_data_balanced_invalid_overlap():
    """Should reject invalid percent_overlap."""
    with pytest.raises(ValueError):
      Data_Balanced_Cover(3, overlap)
```

Parameterizing the test like this makes each one it's own test. This helps identify which case exactly triggered the failure.